### PR TITLE
[curl] Update to 8.9.0

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/curl/dependencies.patch [hunk 1]
```diff
@@ -23,43 +23,36 @@ index 9adb96e0a..9bf69004d 100644
...
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -164,7 +164,8 @@ set(CURL_LIBS "")
+@@ -163,8 +163,9 @@ set(LIBCURL_PC_REQUIRES_PRIVATE "")
  
  if(ENABLE_ARES)
    set(USE_ARES 1)
 -  find_package(CARES REQUIRED)
-+  find_package(CARES NAMES c-ares CONFIG REQUIRED)
-+  set(CARES_LIBRARY c-ares::cares)
```
Comment (dg0yt) [inline ports/curl/dependencies.patch:35]: Why do you replace this with pkg-config?

File: ports/curl/dependencies.patch [hunk 2]
```diff
@@ -23,43 +23,36 @@ index 9adb96e0a..9bf69004d 100644
...
+@@ -591,10 +592,11 @@ endif()
  option(CURL_BROTLI "Set to ON to enable building curl with brotli support." OFF)
  set(HAVE_BROTLI OFF)
  if(CURL_BROTLI)
 -  find_package(Brotli REQUIRED)
-+  find_package(BROTLI NAMES unofficial-brotli REQUIRED)
-+  set(BROTLI_INCLUDE_DIRS "")
-+  set(BROTLI_LIBRARIES "unofficial::brotli::brotlidec")
++  find_package(PkgConfig REQUIRED)
++  pkg_check_modules(BROTLI REQUIRED libbrotlidec)
```
Comment (dg0yt) [inline ports/curl/dependencies.patch:47]: Why do you replace this with pkg-config?

**Review by dg0yt — COMMENTED** [review]
pkg-config and pkg_check_modules in CMake do have a number of issues.

**Comment by talregev** [comment]
I changed because it failed on cmake test port on Linux

**Comment by talregev** [comment]
> pkg-config and pkg_check_modules in CMake do have a number of issues.

Do you referring cross compilation?
curl also shift going through detect libs via pkg-config.
https://github.com/curl/curl/pull/14140

**Comment by dg0yt** [comment]
> I changed because it failed on cmake test port on Linux

Fix the original problem, don't switch to a fragile workaround.

> curl also going to detect libs via pkg-config in cmake.
> [curl/curl#14140](https://github.com/curl/curl/pull/14140)

That PR is about using pkg-config for determining installation prefixes and might deserve its own discussion.

**Comment by talregev** [comment]
> Fix the original problem, don't switch to a fragile workaround.

You can have next PR and fix the original problem.

> That PR is about using pkg-config for determining installation prefixes and might deserve its own discussion.

Please discuss it with victor on his PR.

**Comment by dg0yt** [comment]
> > Fix the original problem, don't switch to a fragile workaround.
> 
> You can have next PR and fix the original problem.

We want linear improvement, not PR ping-pong.

**Comment by talregev** [comment]
> > > Fix the original problem, don't switch to a fragile workaround.
> > 
> > 
> > You can have next PR and fix the original problem.
> 
> We want linear improvement, not PR ping-pong.

Another option, you can take this PR and solve the original problem, and I put this PR on draft, and close it after your merge.

**Comment by talregev** [comment]
I put my PR in draft in favor of @dg0yt PR:
https://github.com/microsoft/vcpkg/pull/40081/commits

**Pull request closed without merging**